### PR TITLE
LPS-158440 Add web storage wrappers with user consent checks

### PR DESCRIPTION
**Pull request by tinycarol**
### References

- [LPS-158440](https://issues.liferay.com/browse/LPS-158440)

### What is the goal?

* Add web storage wrappers with user consent checks
   * These wrappers can be accessed through `Liferay.Util.LocalStorage` and `Liferay.Util.SessionStorage`, or imported as ESM with `import {localStorage, sessionStorage} from 'frontend-js-web'`
   * They have [the same methods as window.[local|session]storage](https://developer.mozilla.org/en-US/docs/Web/API/Storage)

### What does it look like?
imagen

### How to replicate
* Launch Liferay
* Open devtools, see that `Liferay.Util` has two new keys: `LocalStorage` and `SessionStorage`

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/storage.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		this.storage = storage;
+	}
+
+	clear() {
+		return this.storage.clear();
+	}
+
+	getItem(key, type) {
+		if (!checkConsent(type)) {
+			this.removeItem(key);
```
Comment (carloslancha) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/storage.js:28]: We should replicate the same behavior in cookies, if we don't have consent we just don't return the item but we don't remove it

**Review by carloslancha — CHANGES_REQUESTED** [review]
Just a minor change :)

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/storage.js (hunk 1 above)
Comment (tinycarol) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/storage.js]: Done! Changed it in https://github.com/liferay-frontend/liferay-portal/pull/2658/commits/7d3ccb254f44845797fa7d197ba75f093c60bb19, also added a check for this in a test.

**Review by carloslancha — APPROVED** [review]

**Comment by carloslancha** [comment]
ci:forward

**Comment by tinycarol** [comment]
ci:forward

**Pull request closed without merging**